phoenix-core/src/main/java/org/apache/phoenix/query/QueryServicesHelper.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.phoenix.exception.PhoenixNonRetryableRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryServicesHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryServicesHelper.class);
+
+    // Singleton protected by synchronized method
+    static GuidePostsCacheFactory guidePostsCacheFactory = null;
+
+    @VisibleForTesting
+    synchronized GuidePostsCacheFactory loadAndGetGuidePostsCacheFactory(String classString) {
+        Preconditions.checkNotNull(classString);
+        if (guidePostsCacheFactory == null) {
+            try {
+                Class clazz = ClassLoader.getSystemClassLoader().loadClass(classString);
+                Object o = clazz.newInstance();
+                if (!(o instanceof GuidePostsCacheFactory)) {
+                    String msg = String.format("Class %s not an instance of GuidePostsCacheFactory", classString);
+                    LOGGER.error(msg);
+                    throw new PhoenixNonRetryableRuntimeException(msg);
+                }
+                guidePostsCacheFactory = (GuidePostsCacheFactory)o;
```

Comment:
Lets add a log line here indicating the type of guide post cache instantiation with the connection URL. Lets not rely on the fact that every guide post impl prints a log line.

Reply:
I added a log line for the classString but don't want to leak the URL into this class.